internal/plugin/plugin.go

```diff
@@ -0,0 +1,33 @@
+package plugin
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/urfave/cli"
+	"go.step.sm/cli-utils/step"
+)
+
+// LookPath searches for an executable named step-<name>-plugin in the $(step
+// path)/plugins directory or in the directories named by the PATH environment
+// variable.
+func LookPath(name string) (string, error) {
+	fileName := "step-" + name + "-plugin"
+	path := filepath.Join(step.BasePath(), "plugins", fileName)
+	if _, err := os.Stat(path); err == nil {
+		return path, nil
+	}
+	return exec.LookPath(fileName)
+}
+
+// Run starts the given command with the arguments in the context and waits for
+// it to complete.
+func Run(ctx *cli.Context, file string) error {
+	args := ctx.Args()
+	cmd := exec.Command(file, args[1:]...)
```

Comment:
This is quite problematic, especially if you want to compile the plugins yourself. If at some point we add non-opensource plugins this can be an option.

Reply:
Agree with you on it being harder to use when developing new plugins. Having an official distribution channel, for example using `step plugins install`, that does perform verification makes sense to me, though. 

It's also early for thinking about this, but I think there'll be users that are going to require this.